gadgets/advise_seccomp/test/unit/advise_seccomp_test.go

```diff
@@ -0,0 +1,178 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/require"
+
+	gadgettesting "github.com/inspektor-gadget/inspektor-gadget/gadgets/testing"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+)
+
+const adviseDsName = "advise"
+
+type seccompPolicy struct {
+	DefaultAction string   `json:"defaultAction"`
+	Architectures []string `json:"architectures"`
+	Syscalls      []string `json:"syscalls"`
+}
+
+type testDef struct {
+	runnerConfig   *utilstest.RunnerConfig
+	mntnsFilterMap func(info *utilstest.RunnerInfo) *ebpf.Map
+	validate       func(t *testing.T, info *utilstest.RunnerInfo, policies map[string]seccompPolicy)
+}
+
+func TestAdviseSeccompGadget(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+
+	// see https://github.com/inspektor-gadget/inspektor-gadget/issues/3751
+	gadgettesting.MinimumKernelVersion(t, "5.6")
+
+	testCases := map[string]testDef{
+		"all_containers": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			validate: func(t *testing.T, info *utilstest.RunnerInfo, policies map[string]seccompPolicy) {
+				require.GreaterOrEqual(t, len(policies), 1)
+			},
+		},
+		"filtering_no_match": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			mntnsFilterMap: func(info *utilstest.RunnerInfo) *ebpf.Map {
+				return utilstest.CreateMntNsFilterMap(t, 0)
+			},
+			validate: func(t *testing.T, info *utilstest.RunnerInfo, policies map[string]seccompPolicy) {
+				require.Len(t, policies, 0)
+			},
+		},
+		"filtering_exact_match": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			mntnsFilterMap: func(info *utilstest.RunnerInfo) *ebpf.Map {
+				return utilstest.CreateMntNsFilterMap(t, info.MountNsID)
+			},
+			validate: func(t *testing.T, info *utilstest.RunnerInfo, policies map[string]seccompPolicy) {
+				policy, ok := policies[fmt.Sprintf("mntnsid %d", info.MountNsID)]
+				require.True(t, ok)
+				require.Contains(t, policy.Syscalls, "getpid")
+				require.Contains(t, policy.Syscalls, "getppid")
+				require.Contains(t, policy.Syscalls, "getuid")
+				require.Contains(t, policy.Syscalls, "geteuid")
+				require.Contains(t, policy.Syscalls, "openat")
+				require.Contains(t, policy.Syscalls, "close")
+				require.Contains(t, policy.Syscalls, "sysinfo")
+				require.Contains(t, policy.Syscalls, "chdir")
+				require.Contains(t, policy.Syscalls, "mmap")
+				require.Contains(t, policy.Syscalls, "munmap")
```

Comment:
Could it use NotContains for some syscalls that were not used like `reboot`, `unshare` or `setns`?